### PR TITLE
Documentation for view student/class feature in DG

### PR DESCRIPTION
**Pull request by dt-td**

### CODE REVIEW

File: docs/diagrams/ViewStudentSequenceDiagram.puml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+box Model MODEL_COLOR_T1
+participant ":Model" as Model MODEL_COLOR
+participant ":ModelManager" as ModelManager MODEL_COLOR
+end box
+
+box Ui UI_COLOR_T1
+participant "ui:UIManager" as UIManager UI_COLOR
+end box
+
+[-> LogicManager: execute(view -s 1)
```
Comment (JunWei3112) [inline docs/diagrams/ViewStudentSequenceDiagram.puml:19]: Should `view -s 1` be a String instead? This applies to the `parseCommand` method too

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -176,6 +177,41 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
+
+This operation is exposed in the `Model` interface as `Model#viewStudent()` and `Model#viewLesson()`.
+
+Given below is an example usage scenario and how the view student mechanism behaves at each step.
+
+Step 1. The user launches the application for the first time. The `ModelManager` will be initialized with the initial model manager state, and the `currentStatePointer` pointing to that single model manager state.
+
+![ViewStudentState0](images/ViewStudentState0.png)
+
+Step 2. The user executes `view -s 1` command to view the 1st student in TutorAid. The `view` command calls `Model#viewStudent()`, causing the modified state of model manager after the `view -s 1` command executes to be saved in the `modelManagerStateList`, and the `currentStatePointer` pointing to that model manager state.
```
Comment (JunWei3112) [inline docs/DeveloperGuide.md:197]: Should `Model#viewStudent()` be called by the `view -s` command instead?

Comment (dt-td) [reply docs/DeveloperGuide.md]: Right yes I have updated it thank you.

File: docs/diagrams/ViewStudentSequenceDiagram.puml (hunk 1 above)
Comment (dt-td) [reply docs/diagrams/ViewStudentSequenceDiagram.puml]: Yes I have updated it thank you.

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -176,6 +177,41 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
 This section describes some noteworthy details on how certain features are implemented.
 
+### \[Proposed\] View student/lesson feature
```
[the remark this answers is not in the document]
Comment (dt-td) [reply docs/DeveloperGuide.md]: Yes I have removed it. Thank you.

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -13,6 +13,7 @@ title: Developer Guide
    - [Storage component](#storage-component)
    - [Common classes](#common-classes)
 5. [Implementation](#implementation)
+   - [[Proposed] View student/lesson feature](#view-studentlesson-feature)
```
Comment (JunWei3112) [inline docs/DeveloperGuide.md:16]: The `Proposed` tag here should be removed

Comment (dt-td) [reply docs/DeveloperGuide.md]: Right yes , forgot to remove it for the content page. Thx!

**Pull request merged**